drivers/volume/volume.go

```diff
@@ -150,7 +150,10 @@ type Driver interface {
 	ValidateVolumeSnapshotRestore(vol string, snapData *snapv1.VolumeSnapshotData, timeStart time.Time) error
 
 	// CollectDiags collects live diags on a node
-	CollectDiags(n node.Node) error
+	CollectDiags(n node.Node, config *DiagRequestConfig, diagOps DiagOps) error
+
+	// CollectAsyncDiags collects async diags on a node
+	CollectAsyncDiags(n node.Node, config *DiagRequestConfig) error
```

Comment:
From a volume driver interface, we can collapse both of these into the existing `CollectDiags` method and just add a bool Async flag in the DiagRequestConfig. Based on that, internally in portworx.go, you can fork into individual methods. Better to have less public APIs in the interface.